lib/src/pub.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:grinder/grinder.dart';
+import 'package:path/path.dart' as p;
+
+import 'utils.dart';
+
+/// The entire contents of pub's `~/.pub-cache/credentials.json` file.
+///
+/// **Do not check this in directly.** This should only come from secure
+/// sources.
+///
+/// By default this comes from the `PUB_CREDENTIALS` environment variable.
+String get pubCredentials {
+  _pubCredentials ??= Platform.environment["PUB_CREDENTIALS"];
```

Comment:
nit: Strings in the file are mostly defined with single quotes but there are a few lines with double quotes.

Reply:
The Dart style guide explicitly doesn't say anything about string quotes because worrying about it isn't worth the effort :stuck_out_tongue_closed_eyes:.